### ISSUE TITLE
asscalar() fails when set_np_shape(True)

### ISSUE THREAD

**Issue by zheng-da** [apache/incubator-mxnet#18287]
## Description
When we enable `set_np_shape(True)`, I cannot convert a scalar NDArray to python scalar with `asscalar()`. I'm using MXNet 1.6.0.

### Error Message

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-acd043ab0fb2> in <module>
----> 1 a1.asscalar()

/usr/local/lib/python3.6/dist-packages/mxnet/ndarray/ndarray.py in asscalar(self)
   2550         """
   2551         if self.shape != (1,):
-> 2552             raise ValueError("The current array is not a scalar")
   2553         return self.asnumpy()[0]
   2554 

ValueError: The current array is not a scalar
```
## To Reproduce
```python
import mxnet as mx
arr = mx.nd.ones((4,))
mx.set_np_shape(True)
a1, a2, a2, a4 = arr
a1.asscalar()
```

**Comment by szha** [comment]
@zheng-da I verified that this bug is already fixed on master. Do you require this in 1.7?

**Comment by zheng-da** [comment]
yes, we'll wait for mxnet 1.7 release. i'll close the PR.

**Comment by szha** [comment]
@zheng-da reopening as I verified that this bug still persists in 1.7. cc @ciyongch

**Comment by szha** [comment]
The fix seems to be in https://github.com/apache/incubator-mxnet/pull/17672

**Comment by ciyongch** [comment]
Thanks @szha for tagging me, I just tried with https://github.com/apache/incubator-mxnet/pull/17672  on 1.7.x branch, then the issue is gone. I will ping the author to help back-porting the PR to v1.7.x.

**Comment by ciyongch** [comment]
Thanks @Alicia1529 to help backport the fix to v1.7.x and v1.x branches. This issue should be fixed on v1.7.x now @zheng-da .
